tracer/test/Datadog.Trace.TestHelpers.AutoInstrumentation/CrashReport.cs

```diff
@@ -0,0 +1,146 @@
+        }
+
+        return result;
+    }
+
+    private async Task<string> ResolveSymbol(JToken frame)
+    {
+        var name = frame["function"].ToString();
+
+        if (frame["build_id_type"]?.ToString() == "PDB")
+        {
+            var buildId = frame["build_id"].ToString();
+            var pdbName = $"{Path.GetFileNameWithoutExtension(name)}.pdb";
+
+            var moduleAddress = ulong.Parse(frame["module_base_address"].ToString().Replace("0x", string.Empty), NumberStyles.HexNumber);
+            var address = ulong.Parse(frame["ip"].ToString().Replace("0x", string.Empty), NumberStyles.HexNumber);
+
+            var resolvedName = await ResolvePdbSymbol(pdbName, buildId, moduleAddress, address);
+
+            if (resolvedName != null)
+            {
+                var modulePath = name.Split('!')[0];
+                var module = Path.GetFileNameWithoutExtension(modulePath);
+
+                name = $"{module}!{resolvedName}";
+            }
+        }
+
+        return name;
+    }
+
+    private async Task<string> ResolvePdbSymbol(string pdbName, string pdbHash, ulong moduleAddress, ulong address)
+    {
+        var pdbFolder = Path.Combine(_pdbSymbolsFolder, pdbName, pdbHash);
+        var pdbPath = Path.Combine(pdbFolder, pdbName);
+
+        if (!_pdbSymbols.TryGetValue(pdbPath, out var symbols))
+        {
+            var fileExists = File.Exists(pdbPath);
+
+            if (!fileExists)
+            {
+                fileExists = await DownloadSymbol(pdbName, pdbHash, pdbFolder);
+            }
+
+            if (fileExists)
+            {
+                using var reader = new SharpPdb.Native.PdbFileReader(pdbPath);
+
+                symbols = reader.Functions
+                                .OrderBy(f => f.RelativeVirtualAddress)
+                                .Select(f => (f.RelativeVirtualAddress, f.CodeSize, f.GetUndecoratedName()))
+                                .ToList();
+            }
+            else
+            {
+                symbols = new();
+            }
+
+            _pdbSymbols.Add(pdbPath, symbols);
+        }
+
+        string name = null;
+
+        foreach (var symbol in symbols)
+        {
+            if (symbol.Offset > address - moduleAddress)
+            {
+                break;
+            }
+
+            name = $"{symbol.Symbol}+{address - moduleAddress - symbol.Offset:x2}";
+
+            if (symbol.Offset + symbol.Size < address - moduleAddress)
+            {
+                name = $"<unknown> ({name})";
+            }
+        }
+
+        return name;
+    }
+
+    private async Task<bool> DownloadSymbol(string pdbName, string pdbHash, string destinationFolder)
+    {
+        var serverUrl = "https://msdl.microsoft.com/download/symbols";
+        var url = $"{serverUrl}/{pdbName}/{pdbHash}/{pdbName}";
+
+        _output.WriteLine($"Downloading symbol from {url} to {destinationFolder}");
+
+        using var client = new HttpClient();
```

Comment:
Probably fine, but given we're going to be potentially calling this a bunch of times, might want to make this a field on the instance instead, so we don't end up disposing a bunch of them and risking socket exhaustion. v unlikely I know, but still easy to be on the safe side 🤷‍♂️

Reply:
Meh but as you wish